### PR TITLE
Add Discourse module

### PR DESCRIPTION
**Pull request by sowasred2012**
## Done

Added discourse docs module, along with documentation layout scss and markup

## QA
- Check out this branch
- `./run` the project
- Visit http://0.0.0.0:8026/docs
- See that the content there is being pulled from https://discourse.ubuntu.com/t/multipass-documentation-outline/8294

Fixes https://github.com/canonical-web-and-design/web-squad/issues/2146 & https://github.com/canonical-web-and-design/web-squad/issues/2147

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 canonicalwebteam.flask-base==0.3.3
-gunicorn[gevent]
+canonicalwebteam.discourse-docs==0.6.5
+gunicorn[gevent]
```
Comment (nottrobin) [inline requirements.txt]: This is possibly a bit over-cautious, but would you mind placing `gunicorn[gevent]` directly below `flask-base`? If `pip` were a decent package manager it wouldn't matter, but as things are there's a possibility that a sub-dependency from `discourse-docs` could lead to `gevent` not installing its preferred version of a dependency, which could be problematic.

File: templates/_layouts/default.html [hunk 2]
```diff
@@ -6,7 +6,7 @@
   <meta http-equiv="X-UA-Compatible" content="IE=edge">
   <meta name="viewport" content="width=device-width, initial-scale=1">
 
-  <title>Multipass - Orchestrates virtual Ubuntu instances</title>
+  <title>{% block title %}{% endblock %} | Multipass - Orchestrates virtual Ubuntu instances</title>
```
Comment (nottrobin) [inline templates/_layouts/default.html]: This is gonna look pretty funny if the `title` block isn't defined - the title will read:

>  | Multipass - Orchestrates virtual Ubuntu instances

Not sure what the best way around this is. Maybe we should change it to:

```
{% block title %}Orchestrate virtual Ubuntu instances{% endblock %} | Multipass
```

I'm not sure who to run that change by - maybe @pmahnke ?

File: webapp/app.py [hunk 3]
```diff
@@ -18,3 +22,19 @@
 @app.route("/")
 def index():
     return render_template("index.html")
+
+
+url_prefix = "/docs"
+server_docs_parser = DocParser(
+    api=DiscourseAPI(base_url="https://discourse.ubuntu.com/"),
```
Comment (nottrobin) [inline webapp/app.py]: Had a chat with @tbille and we feel it's better to override the cached session (which we're deprecating anyway) with a [Talisker session](https://talisker.readthedocs.io/en/latest/requests.html#using-a-talisker-session) instead, so we get API request latency metrics in the logs.

Could you please instantiate `DiscourseAPI` as follows:

``` python3
import talisker

# ...
    api=DiscourseAPI(
        base_url="https://discourse.ubuntu.com/",
        session=talisker.requests.get_session()
    ),
```

?

**Comment by anthonydillon** [comment]
There is an issue with code causing the view to scroll on small screens.
Screenshot_20191217-121625__01

File: webapp/app.py (hunk 3 above)
Comment (sowasred2012) [reply webapp/app.py]: Will do!

**Comment by Saviq** [comment]
FWIW:

```
<ScottMN> hey, was wondering if I could run this by you - it's the multipass discourse docs, pulled into multipass.run: https://multipass-run-canonical-web-and-design-pr-66.run.demo.haus/docs
<ScottMN> are you happy for this to be released? quite a few of the docs are missing or incomplete, want me to remove those placeholder links or can those posts be created?
<Saviq> I think this is fine, we'll be adding that documentation asap
```

**Comment by sowasred2012** [comment]
@anthonydillon @nottrobin I've addressed your points, this is ready for another review.

**Comment by anthonydillon** [comment]
@sowasred2012 I still see the same issue on my phone. Also there is no selected styling in the navigation.

File: templates/docs/document.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
+
+{% extends "docs/_base_docs.html" %}
+
+{% block title %}{{ document.title }} | Multipass documentation | {% endblock %}
```
Comment (nottrobin) [inline templates/docs/document.html]: I'm not crazy about this solution - requiring creators of new templates to know and remember to add ` | ` to the end of their `title` block seems set up to fail.

If we can't come up with a default title in `_layouts/default.html` that works smoothly with the `title` block, we could use this slightly more complex syntax instead:

```
{% if self.title() %}
  <title>{{ self.title() }}</title>
{% else %}
  <title>Multipass - Orchestrates virtual Ubuntu instances</title>
{% endif %}
```

Comment (nottrobin) [reply templates/docs/document.html]: Although, if you're going with a `Multipass documentation` suffix, it seems like we could probably let the `title` block override the entire thing:

``` html
<!--_layouts/default.html -->

<title>{% block title %}Multipass - Orchestrates virtual Ubuntu instances{% title %}</title>

<!-- docs/document.html -->
{% block title %}{{ document.title }} | Multipass documentation{% endblock %}
```

**Comment by anthonydillon** [comment]
@sowasred2012 thanks, could you update the URL map to not include multipass as its already on the site and doesnt need the duplication. For example:
- `/docs/installing-multipass-on-linux` to `/docs/installing-on-linux`
- `/docs/multipass-launch-command` to `/docs/launch-command`
- `/docs/working-with-multipass-instances` to `/docs/working-with-instances`

**Comment by sowasred2012** [comment]
@anthonydillon good shout, just updated those :+1:

**Pull request merged**